src/controllers/controller.line.js

```diff
@@ -210,17 +208,17 @@ module.exports = DatasetController.extend({
 			key = keys[i];
 			values[key] = resolve([
 				custom[key],
-				dataset[key],
+				datasetOpts[key],
 				elementOptions[key]
 			], context);
 		}
 
 		// The default behavior of lines is to break at null values, according
 		// to https://github.com/chartjs/Chart.js/issues/2435#issuecomment-216718158
 		// This option gives lines the ability to span gaps
-		values.spanGaps = valueOrDefault(dataset.spanGaps, options.spanGaps);
-		values.tension = valueOrDefault(dataset.lineTension, elementOptions.tension);
-		values.steppedLine = resolve([custom.steppedLine, dataset.steppedLine, elementOptions.stepped]);
+		values.spanGaps = resolve([datasetOpts.spanGaps, options.spanGaps]);
+		values.tension = resolve([datasetOpts.tension, elementOptions.tension]);
```

Comment:
Why `resolve`? `valueOrDefault` is faster.

Reply:
done